SMLHelper/Patchers/OptionsPanelPatcher.cs

```diff
@@ -132,7 +142,17 @@ private class StatesConfig
 
                     public HeadingState this[string name]
                     {
-                        get => states.TryGetValue(name, out HeadingState state)? state: HeadingState.Expanded;
+                        get
+                        {
+                            if(string.IsNullOrEmpty(name))
```

Comment:
Why this check is needed here?

Reply:
somehow in BZ a null is getting passed into this getter.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=netframework-4.0

An `ArgumentNullException` can be thrown if it's passed into `Dictionary.TryGetValue`.  
This seems like a sensible check to make.

Reply:
exactly what was happening.

Reply:
Yeah, probably `text` field in new BZ Text component can be null if it is empty.

@MrPurple6411 Let it be something like this then:
`states.TryGetValue(name ?? "", out HeadingState state)? state: HeadingState.Expanded;`

Reply:
ahhhh ok thats the issue... Text is no longer used in BZ  its now TextMeshProUGUI   that would do it. meaning i need to go find your ref to Text that is getting passed to this.

Reply:
It's there https://github.com/SubnauticaModding/SMLHelper/blob/f59e6ec623fd39fe3697a6e676306c0a4125d994/SMLHelper/Patchers/OptionsPanelPatcher.cs#L219